internal/util/function/text_embedding_function.go

```diff
@@ -0,0 +1,56 @@
+/*
+ * # Licensed to the LF AI & Data foundation under one
+ * # or more contributor license agreements. See the NOTICE file
+ * # distributed with this work for additional information
+ * # regarding copyright ownership. The ASF licenses this file
+ * # to you under the Apache License, Version 2.0 (the
+ * # "License"); you may not use this file except in compliance
+ * # with the License. You may obtain a copy of the License at
+ * #
+ * #     http://www.apache.org/licenses/LICENSE-2.0
+ * #
+ * # Unless required by applicable law or agreed to in writing, software
+ * # distributed under the License is distributed on an "AS IS" BASIS,
+ * # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * # See the License for the specific language governing permissions and
+ * # limitations under the License.
+ */
+
+package function
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+)
+
+const (
+	Provider string = "provider"
+)
+
+const (
+	OpenAIProvider string = "openai"
+)
+
+func getProvider(schema *schemapb.FunctionSchema) (string, error) {
+	for _, param := range schema.Params {
+		switch strings.ToLower(param.Key) {
+		case Provider:
+			return strings.ToLower(param.Value), nil
+		default:
+		}
+	}
+	return "", fmt.Errorf("The provider parameter was not found in the function's parameters")
```

Comment:
richer info in error: the embedding service provider parameter was not found ...